eng/scripts/Merge-Branch.ps1

```diff
@@ -37,20 +37,20 @@ function ErrorExit($exitCode) {
 
 # start a merge, but leave it open
 Write-Verbose "git merge $SourceBranch --no-ff --no-commit"
-git merge $SourceBranch --no-ff --no-commit
-if ($LASTEXITCODE) { ErrorExit $LASTEXITCODE }
+git merge $SourceBranch --no-ff --no-commit | Tee-Object -Variable mergeOutput
+if ($LASTEXITCODE -and -not $mergeOutput.EndsWith('Automatic merge failed; fix conflicts and then commit the result.')) { ErrorExit $LASTEXITCODE }
```

Comment:
Do we need to fail below if there is a merge failure still?

Reply:
If there's a failure other than the expected "fix conflicts" we should error exit with the "Unknown state, git reset --hard" message

If there's an exit code, but git's telling us to fix our conflicts and commit, then we're in an expected state and we should just continue as usual